streams/shared/src/main/scala/zio/stream/experimental/ZSink.scala

```diff
@@ -346,7 +346,7 @@ object ZSink {
    * A sink that collects all of its inputs into chunks of maximum size `n`.
    */
   def collectAllN[Err, In](n: Int): ZSink[Any, Err, In, Err, In, Chunk[In]] =
-    foldUntil(Chunk.empty: Chunk[In], n.toLong)(_ appended _)
+    foldUntil(Chunk.empty: Chunk[In], n.toLong)(_ :+ _)
```

Comment:
Should we be using a `ChunkBuilder` here?

Reply:
Would be neat, but it's one extra step to materialize it.

Reply:
Ah a ChunkBuilder is a good point. We can also hint it with the size so should be significantly more efficient.

Reply:
Tackled in 0427c094c5af0a5bbe66f122582c65363bede6e3.

Reply:
Here's the updated version: 375647a70a2149f6950e378174a04d390e47fdd9